tests/fixtures/envs/dummy/dummy_discrete_pixel_env.py

```diff
@@ -1,31 +1,60 @@
+from unittest import mock
+
 import gym
 import numpy as np
 
 from tests.fixtures.envs.dummy import DummyEnv
 
 
 class DummyDiscretePixelEnv(DummyEnv):
-    """A dummy discrete environment."""
+    """
+    A dummy discrete pixel environment.
+
+    It follows Atari game convention, where actions are 'NOOP', 'FIRE', ...
+    It also contains self.unwrapped.ale.lives, get_action_meanings for testing.
+    Observations are as following:
+        after reset    : np.zeros(self._shape).
+        action 1 (FIRE): np.ones(self._shape).
+        otherwise      : random if self.random is True,
+            otherwise previous state + action.
+    """
 
     def __init__(self, random=True):
         super().__init__(random)
+        self.unwrapped.get_action_meanings = self._get_action_meanings
+        self.unwrapped.ale = mock.Mock()
+        self.unwrapped.ale.lives = self.get_lives
+        self._shape = (10, 10, 3)
+        self._observation_space = gym.spaces.Box(
+            low=0, high=255, shape=self._shape, dtype=np.uint8)
+        self.step_called = 0
```

Comment:
this should be _private

Reply:
why is this variable here? it seems only for testing?

you can use mock to count how many times something is called